### PR TITLE
FINERACT-1770: Introduce Lombok to sms module

### PR DESCRIPTION
**Pull request by hmg65**
## Description

Describe the changes made and why they were made.

Ignore if these details are present on the associated [Apache Fineract JIRA ticket](https://github.com/apache/fineract/pull/1284).

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [x] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [x] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/legacy-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/domain/SmsMessage.java [hunk 1]
```diff
@@ -86,39 +93,26 @@ public class SmsMessage extends AbstractPersistableCustom {
...
+                .setSmsCampaign(smsCampaign).setNotification(isNotification);
     }
 
     public static SmsMessage instance(String externalId, final Group group, final Client client, final Staff staff,
             final SmsMessageStatusType statusType, final String message, final String mobileNo, final SmsCampaign smsCampaign,
             final boolean isNotification) {
 
-        return new SmsMessage(externalId, group, client, staff, statusType, message, mobileNo, smsCampaign, isNotification);
-    }
+        return new SmsMessage().setExternalId(externalId).setGroup(group).setClient(client).setStaff(staff)
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/domain/SmsMessage.java:113]: Missing: `this.submittedOnDate = DateUtils.getBusinessLocalDate();`

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/domain/SmsMessage.java [hunk 2]
```diff
@@ -86,39 +93,26 @@ public class SmsMessage extends AbstractPersistableCustom {
...
+                .setStatusType(SmsMessageStatusType.PENDING.getValue()).setMessage(message).setMobileNo(mobileNo)
+                .setSmsCampaign(smsCampaign).setNotification(isNotification);
     }
 
     public static SmsMessage sentSms(final String externalId, final Group group, final Client client, final Staff staff,
             final String message, final String mobileNo, final SmsCampaign smsCampaign, final boolean isNotification) {
-        return new SmsMessage(externalId, group, client, staff, SmsMessageStatusType.WAITING_FOR_DELIVERY_REPORT, message, mobileNo,
-                smsCampaign, isNotification);
+        return new SmsMessage().setExternalId(externalId).setGroup(group).setClient(client).setStaff(staff)
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/domain/SmsMessage.java:104]: Missing: `this.submittedOnDate = DateUtils.getBusinessLocalDate();`

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/domain/SmsMessage.java [hunk 3]
```diff
@@ -86,39 +93,26 @@ public class SmsMessage extends AbstractPersistableCustom {
 
     public static SmsMessage pendingSms(final String externalId, final Group group, final Client client, final Staff staff,
             final String message, final String mobileNo, final SmsCampaign smsCampaign, final boolean isNotification) {
-        return new SmsMessage(externalId, group, client, staff, SmsMessageStatusType.PENDING, message, mobileNo, smsCampaign,
-                isNotification);
+        return new SmsMessage().setExternalId(externalId).setGroup(group).setClient(client).setStaff(staff)
```
Comment (galovics) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/domain/SmsMessage.java:97]: Missing: `this.submittedOnDate = DateUtils.getBusinessLocalDate();`

**Review by galovics — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/domain/SmsMessage.java (hunk 2 above)
Comment (hmg65) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/domain/SmsMessage.java:104]: done

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/domain/SmsMessage.java (hunk 1 above)
Comment (hmg65) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/domain/SmsMessage.java:113]: done

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/domain/SmsMessage.java (hunk 3 above)
Comment (hmg65) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/domain/SmsMessage.java:97]: done

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/data/SmsData.java [hunk 4]
```diff
@@ -18,75 +18,33 @@
...
-    private final Long id;
-    private final Long groupId;
-    private final Long clientId;
-    private final Long staffId;
-    private final EnumOptionData status;
-    private final String mobileNo;
-    private final String message;
-    private final Long providerId;
-    private final String campaignName;
+    private Long id;
```
Comment (BLasan) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/data/SmsData.java:35]: Why not using `@RequiredArgsConstructor` instead ?

**Review by BLasan — CHANGES_REQUESTED** [review]

**Review by galovics — APPROVED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/data/SmsData.java (hunk 4 above)
Comment (hmg65) [reply fineract-provider/src/main/java/org/apache/fineract/infrastructure/sms/data/SmsData.java:35]: @BLasan we do not want to keep `final` keyword.

**Review by BLasan — APPROVED** [review]

**Pull request merged**